File: 北理珠/19天体杯复盘/L1-6.cpp
#include <bits/stdc++.h>
using namespace std;

int T,n,x,y; 

int calc(int x)
{
	int ret = 0;
	while (x)
	{
		ret += x%10;
		x /= 10;
	}
	return ret;
}

int main()
{
	cin >> T;
	while (T--)
	{
		cin >> n;
		x = n/1000;
		y = n%1000;
		if (calc(x) == calc(y)) cout << "You are lucky!" << endl;
		else cout << "Wish you good luck." << endl; 
	}
}

File: ACM/22天梯赛/L1-6.py
def calc(s):
    ret = ''
    for i in range(1,len(s)):
        if (int(s[i])%2==int(s[i-1])%2):
            ret += max(s[i],s[i-1])
    return ret

A = calc(input())
B = calc(input())
if (A==B):
    print(A)
else:
    print(A)
    print(B)